cmd/minishift/cmd/services/proxy.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"github.com/minishift/minishift/cmd/minishift/cmd/config"
+	"github.com/minishift/minishift/pkg/minishift/network/proxy"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+const (
+	proxyServerPortFlag = "port"
+	proxyUpstreamFlag   = "upstream"
+)
+
+var (
+	proxyServerPortFromFlag   int
+	proxyUpstreamAddrFromFlag string
+
+	servicesProxyCmd = &cobra.Command{
+		Use:    "proxy",
+		Short:  "Starts a proxy server on host",
+		Long:   `Starts a proxy server on host`,
+		Run:    runProxy,
+		Hidden: true,
+	}
+)
+
+func init() {
+	servicesProxyCmd.Flags().IntVarP(&proxyServerPortFromFlag, proxyServerPortFlag, "p", 3128, "The server port.")
```

Comment:
Why the proxy command is not added to `ServicesCmd` but instead you created a new `servicesProxyCmd`?

Reply:
becauese it is about the flag for the proxy (?) and not generally about the services.

Reply:
@gbraad I still do not understand the downside  of linking sub command to the parent command when the command is hidden anyway. `$ minishift services proxy` is the command anyway if any developer wants to run it.

Reply:
I am not following your way of thinking...

These are commands invoked by minishift as daemons. They have no real purpose to be started by the user. They are JUST grouped here.